src/java/org/apache/cassandra/io/sstable/AbstractSSTableSimpleWriter.java

```diff
@@ -110,6 +112,39 @@ private static SSTableId getNextId(File directory, final String columnFamily) th
         }
     }
 
+    /**
+     * Returns the partition update builder for the given key
+     * @param key the partition key of the partition update
+     * @return an update on partition {@code key} that is tied to this writer
+     */
+    protected PartitionUpdate.Builder createUpdateBuilder(DecoratedKey key)
+    {
+        return createSizeCountingPartitionUpdateBuilder(key, row -> {}, () -> {});
+    }
+
+    /**
+     * Returns the partition update builder that counts row sizes for the given key.
+     * @param key the partition key of the partition update
+     * @param onRow consumes the row to count the size
+     * @param onRowAdded runnale that runs after the row is added
+     * @return an update on partition {@code key} that is tied to this writer
+     */
+    protected PartitionUpdate.Builder createSizeCountingPartitionUpdateBuilder(DecoratedKey key,
+                                                                               Consumer<Row> onRow,
+                                                                               Runnable onRowAdded)
```

Comment:
to me it feels that all the logic can be encapsulated in the consumer lambda, and renders the runnable unnecessary. For the only calling site where we actually use the Runnable parameter I would refactor it as:

```
previous = createSizeCountingPartitionUpdateBuilder(key, row -> {
                countRow(row);
                maybeSync();
            });
```

```suggestion
    protected PartitionUpdate.Builder createSizeCountingPartitionUpdateBuilder(DecoratedKey key, Consumer<Row> onRow)
```